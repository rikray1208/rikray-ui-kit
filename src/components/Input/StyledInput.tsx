import styled, {css} from "../../utils/styled-components";
import {InputProps} from "./index";
import {Size, Status} from "../types";

function setBorderRadius(prefix: string | undefined, postfix: string | undefined) {
    if (prefix && postfix) {
        return css`
            border-radius: 0;
        `
    }
    if(!prefix && !postfix) {
        return css`
          border-radius: ${({theme}) => theme.rounded.SemiRound};
        `
    }
    if (prefix) {
        return css`
          border-radius: ${({theme}) => theme.rounded.SemiRound + ' ' + '0' + ' ' + '0' + ' ' + theme.rounded.SemiRound };
        `
    }
    if (postfix){
        return css`
          border-radius: ${({theme}) => '0' + ' ' + theme.rounded.SemiRound + ' ' + theme.rounded.SemiRound + ' ' + '0'};
          
        `
    }
}

const CommonTabs = css<InputProps>`
  ${({theme, Size, type}) => css`
    padding: ${theme.Spacing.padding["2"]} ${theme.Spacing.padding["4"]};
    margin: 0;

    font-size: ${theme.Typography.fontSize[Size as Size]};
    line-height: ${theme.Typography.lineHeight[Size as Size]};
    
    background-color: white;
    border-top: 1px solid;
    border-bottom: 1px solid;
    border-color: ${type == 'base' ? theme.colors['grey'] : theme.colors[type as Status]};
    color: black;
  `}
`
export const StyledPrefix = styled.div<InputProps>`
  ${({theme}) => css`
    border-left: 1px solid;
    border-radius: ${theme.rounded.SemiRound + ' ' + '0' + ' ' + '0' + ' ' + theme.rounded.SemiRound};

    ${CommonTabs}
  `}
`

export const StyledPostfix = styled.div<InputProps>`
  ${({theme}) => css`
    border-right: 1px solid;
    border-radius: ${'0' + ' ' + theme.rounded.SemiRound + ' ' + theme.rounded.SemiRound + ' ' + '0'};
    
    ${CommonTabs}
  `}
`

const InputStyles = css<InputProps>`
  ${({theme, type, Size, postfix, prefix}) =>
          css`
            display: inline-block;
            min-width: 0;
            width: 100%;
            
            padding: ${theme.Spacing.padding["2"]} ${theme.Spacing.padding["4"]};
            margin: 0;

            font-size: ${theme.Typography.fontSize[Size as Size]};
            line-height: ${theme.Typography.lineHeight[Size as Size]};

            transition: ${theme.transitions["input"]};

            background-color: white;
            border-width: 1px;
            border-style: solid;
            border-color: ${type == 'base' ? theme.colors['grey'] : theme.colors[type as Status]};
            color: black;
            outline: none;
            
            ${setBorderRadius(postfix, prefix)}
            
            &:hover {
              border-color: ${type == 'base' ? theme.colors["primary"] : theme.colors[type as Status]};
            }

            &:focus {
              border-color: ${type == 'base' ? theme.colors["primary"] : theme.colors[type as Status]};
              filter: ${theme.Filter.dropShadow["sm"]};
            }

            &::placeholder {
              color: black;
              filter: opacity(${theme.Filter.opacity["placeholder"]});
            }
          `
  }
`
export const StyledInputWrapper = styled.div`
  display: flex;
  width: 100%;
`

const StyledInput = styled.input<InputProps>`
    ${InputStyles}
`;

export default StyledInput
