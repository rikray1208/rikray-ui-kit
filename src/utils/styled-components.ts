import * as styledComponents from 'styled-components';
import {ITheme} from "../theme/types"


const {
    default: styled,
    css,
    createGlobalStyle,
    keyframes,
    ThemeProvider,
} = styledComponents as styledComponents.ThemedStyledComponentsModule<ITheme>;

export { css, createGlobalStyle, keyframes, ThemeProvider };
export default styled;
