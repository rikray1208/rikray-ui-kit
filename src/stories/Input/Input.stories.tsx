import {Args, ComponentMeta, ComponentStory} from "@storybook/react";
import Input from "../../components/Input";
import React from "react";

export default {
    title: 'Design System/Inputs/Input',
    component: Input
} as ComponentMeta<typeof Input>;

export const decorators = [
    (Story: any) => (
        <div style={{ margin: '3em' }}>
            <Story />
        </div>
    ),
];

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />;

export const SmallInput = Template.bind({});
SmallInput.decorators = decorators
SmallInput.args = {
    placeholder: 'Placeholder...',
    Size: 'Small'
}

export const MediumInput = Template.bind({});
MediumInput.decorators = decorators
MediumInput.args = {
    placeholder: 'Placeholder...',
    Size: 'Medium'
}

export const LargeInput = Template.bind({});
LargeInput.decorators = decorators
LargeInput.args = {
    placeholder: 'Placeholder...',
    Size: 'Large'
}

export const PrefixInput = Template.bind({});
PrefixInput.decorators = decorators
PrefixInput.args = {
    placeholder: 'Placeholder...',
    Size: 'Small',
    prefix: 'https://'
}

export const PostfixInput = Template.bind({});
PostfixInput.decorators = decorators
PostfixInput.args = {
    placeholder: 'Placeholder...',
    Size: 'Small',
    postfix: '.com'
}

export const PrefixAndPostfixInput = Template.bind({});
PrefixAndPostfixInput.decorators = decorators
PrefixAndPostfixInput.args = {
    placeholder: 'Placeholder...',
    Size: 'Small',
    prefix: 'https://',
    postfix: '.com'
}
