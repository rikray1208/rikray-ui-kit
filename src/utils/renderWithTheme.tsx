import {render} from "@testing-library/react";
import React from "react";
import {ITheme} from "../theme/types";
import {ThemeProvider} from "./styled-components";

export const renderWithTheme = (component: React.ReactNode, theme: ITheme) => {
    return render(
        <ThemeProvider theme={theme}>
            {component}
        </ThemeProvider>
    )
}
