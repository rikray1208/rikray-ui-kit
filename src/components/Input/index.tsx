import React, {FC, InputHTMLAttributes} from 'react';
import {Size, Status} from "../types";
import StyledInput, {StyledInputWrapper, StyledPostfix, StyledPrefix} from "./StyledInput";
import {ThemeProvider} from "../../utils/styled-components";
import BaseTheme from "../../theme";

export type BaseInputProps = Omit<InputHTMLAttributes<HTMLInputElement>, 'type' | 'size'>

export interface InputProps extends BaseInputProps {
    type?: Status
    placeholder?: string
    Size?: Size
    prefix?: string
    postfix?: string

}

const Input: FC<InputProps> = (props) => {
    return (
        <>
            <ThemeProvider theme={BaseTheme}>
                <StyledInputWrapper>
                    {props.prefix && <StyledPrefix {...props}>{props.prefix}</StyledPrefix>}
                    <StyledInput {...props}/>
                    {props.postfix && <StyledPostfix {...props}>{props.postfix}</StyledPostfix>}
                </StyledInputWrapper>
            </ThemeProvider>
        </>
    );
};

Input.defaultProps = {
    type: 'base',
    Size: "Small",
}

export default Input;
