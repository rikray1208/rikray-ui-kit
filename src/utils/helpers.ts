import {Shape, Size} from "../components/types";
import {css} from "styled-components";
import React from "react";
import renderer from "react-test-renderer";
import {ComponentMap} from "../theme/types";

type Component = keyof  ComponentMap;

export function setShape(shape: Shape, Component: Component) {
    if (shape === 'Round')
        return css`border-radius: ${({theme}) => theme[Component]["borderRadius-lg"] + 'px'};`

    if (shape === 'SemiRound')
        return css`border-radius: ${({theme}) => theme[Component]["borderRadius-sm"] + 'px'};`

    if (shape === 'Rectangle')
        return css`border-radius: ${({theme}) => theme[Component]["borderRadius-xs"] + 'px'};`
}
export function setFontSize(size: Size, Component: Component) {
    if (size == 'Small')
        return css`
          font-size: ${({theme}) => theme[Component]["fontSize-sm"] + 'rem'};
          line-height: ${({theme}) => theme[Component]["lineHeight-sm"] + 'rem'};
        `

    if (size == 'Medium')
        return css`
          font-size: ${({theme}) => theme[Component]["fontSize-base"] + 'rem'};
          line-height: ${({theme}) => theme[Component]["lineHeight-base"] + 'rem'};
        `

    if (size == 'Large')
        return css`
          font-size: ${({theme}) => theme[Component]["fontSize-lg"] + 'rem'};
          line-height: ${({theme}) => theme[Component]["lineHeight-lg"] + 'rem'};
        `
}

export const assertReactElement = (element: React.ReactElement) => {
    const component = renderer.create(element);
    return component.toJSON();
};
