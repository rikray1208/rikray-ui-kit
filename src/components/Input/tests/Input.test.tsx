import React from "react";
import renderer from "react-test-renderer";
import BaseTheme from "../../../theme";
import { ThemeProvider } from "../../../utils/styled-components";
import Input from "../index";
import 'jest-styled-components'

const assertReactElement = (element: React.ReactElement) => {
    const component = renderer.create(element);
    return component.toJSON();
};

describe('Input', () => {
    it('should return a Input', () => {
        const element = (
            <ThemeProvider theme={BaseTheme}><Input/></ThemeProvider>
        )
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Small Input', () => {
        const element = (
            <ThemeProvider theme={BaseTheme}><Input Size={"Small"}/></ThemeProvider>
        )
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Medium Input', () => {
        const element = (
            <ThemeProvider theme={BaseTheme}><Input Size={"Medium"}/></ThemeProvider>
        )
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a Large Input', () => {
        const element = (
            <ThemeProvider theme={BaseTheme}><Input Size={"Large"}/></ThemeProvider>
        )
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a with prefix and postfix Input', () => {
        const element = (
            <ThemeProvider theme={BaseTheme}><Input Size={"Large"} prefix={'https://'} postfix={'.com'}/></ThemeProvider>
        )
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a success Input', () => {
        const element = (
            <ThemeProvider theme={BaseTheme}><Input type={'success'}/></ThemeProvider>
        )
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a warning Input', () => {
        const element = (
            <ThemeProvider theme={BaseTheme}><Input type={'warning'}/></ThemeProvider>
        )
        expect(assertReactElement(element)).toMatchSnapshot();
    })
    it('should return a error Input', () => {
        const element = (
            <ThemeProvider theme={BaseTheme}><Input type={'error'}/></ThemeProvider>
        )
        expect(assertReactElement(element)).toMatchSnapshot();
    })
})
