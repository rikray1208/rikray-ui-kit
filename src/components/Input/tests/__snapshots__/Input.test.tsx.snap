// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should return a Input 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  display: inline-block;
  min-width: 0;
  width: 100%;
  padding: 0.5rem 1rem;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-transition: all .2s cubic-bezier(0.4,0,0.2,1);
  transition: all .2s cubic-bezier(0.4,0,0.2,1);
  background-color: white;
  border-width: 1px;
  border-style: solid;
  border-color: rgb(209 213 219);
  color: black;
  outline: none;
  border-radius: 8px;
}

.c1:hover {
  border-color: #3B74F3;
}

.c1:focus {
  border-color: #3B74F3;
  -webkit-filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
  filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
}

.c1::-webkit-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::-moz-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1:-ms-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

<div
  className="c0"
>
  <input
    className="c1"
    type="base"
  />
</div>
`;

exports[`Input should return a Large Input 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  display: inline-block;
  min-width: 0;
  width: 100%;
  padding: 0.5rem 1rem;
  margin: 0;
  font-size: 1.25rem;
  line-height: 1.75rem;
  -webkit-transition: all .2s cubic-bezier(0.4,0,0.2,1);
  transition: all .2s cubic-bezier(0.4,0,0.2,1);
  background-color: white;
  border-width: 1px;
  border-style: solid;
  border-color: rgb(209 213 219);
  color: black;
  outline: none;
  border-radius: 8px;
}

.c1:hover {
  border-color: #3B74F3;
}

.c1:focus {
  border-color: #3B74F3;
  -webkit-filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
  filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
}

.c1::-webkit-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::-moz-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1:-ms-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

<div
  className="c0"
>
  <input
    className="c1"
    type="base"
  />
</div>
`;

exports[`Input should return a Medium Input 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  display: inline-block;
  min-width: 0;
  width: 100%;
  padding: 0.5rem 1rem;
  margin: 0;
  font-size: 1rem;
  line-height: 1.5rem;
  -webkit-transition: all .2s cubic-bezier(0.4,0,0.2,1);
  transition: all .2s cubic-bezier(0.4,0,0.2,1);
  background-color: white;
  border-width: 1px;
  border-style: solid;
  border-color: rgb(209 213 219);
  color: black;
  outline: none;
  border-radius: 8px;
}

.c1:hover {
  border-color: #3B74F3;
}

.c1:focus {
  border-color: #3B74F3;
  -webkit-filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
  filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
}

.c1::-webkit-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::-moz-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1:-ms-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

<div
  className="c0"
>
  <input
    className="c1"
    type="base"
  />
</div>
`;

exports[`Input should return a Small Input 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  display: inline-block;
  min-width: 0;
  width: 100%;
  padding: 0.5rem 1rem;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-transition: all .2s cubic-bezier(0.4,0,0.2,1);
  transition: all .2s cubic-bezier(0.4,0,0.2,1);
  background-color: white;
  border-width: 1px;
  border-style: solid;
  border-color: rgb(209 213 219);
  color: black;
  outline: none;
  border-radius: 8px;
}

.c1:hover {
  border-color: #3B74F3;
}

.c1:focus {
  border-color: #3B74F3;
  -webkit-filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
  filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
}

.c1::-webkit-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::-moz-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1:-ms-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

<div
  className="c0"
>
  <input
    className="c1"
    type="base"
  />
</div>
`;

exports[`Input should return a error Input 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  display: inline-block;
  min-width: 0;
  width: 100%;
  padding: 0.5rem 1rem;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-transition: all .2s cubic-bezier(0.4,0,0.2,1);
  transition: all .2s cubic-bezier(0.4,0,0.2,1);
  background-color: white;
  border-width: 1px;
  border-style: solid;
  border-color: #ff3333;
  color: black;
  outline: none;
  border-radius: 8px;
}

.c1:hover {
  border-color: #ff3333;
}

.c1:focus {
  border-color: #ff3333;
  -webkit-filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
  filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
}

.c1::-webkit-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::-moz-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1:-ms-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

<div
  className="c0"
>
  <input
    className="c1"
    type="error"
  />
</div>
`;

exports[`Input should return a success Input 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  display: inline-block;
  min-width: 0;
  width: 100%;
  padding: 0.5rem 1rem;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-transition: all .2s cubic-bezier(0.4,0,0.2,1);
  transition: all .2s cubic-bezier(0.4,0,0.2,1);
  background-color: white;
  border-width: 1px;
  border-style: solid;
  border-color: #198754;
  color: black;
  outline: none;
  border-radius: 8px;
}

.c1:hover {
  border-color: #198754;
}

.c1:focus {
  border-color: #198754;
  -webkit-filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
  filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
}

.c1::-webkit-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::-moz-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1:-ms-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

<div
  className="c0"
>
  <input
    className="c1"
    type="success"
  />
</div>
`;

exports[`Input should return a warning Input 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c1 {
  display: inline-block;
  min-width: 0;
  width: 100%;
  padding: 0.5rem 1rem;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.25rem;
  -webkit-transition: all .2s cubic-bezier(0.4,0,0.2,1);
  transition: all .2s cubic-bezier(0.4,0,0.2,1);
  background-color: white;
  border-width: 1px;
  border-style: solid;
  border-color: #ffcc00;
  color: black;
  outline: none;
  border-radius: 8px;
}

.c1:hover {
  border-color: #ffcc00;
}

.c1:focus {
  border-color: #ffcc00;
  -webkit-filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
  filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
}

.c1::-webkit-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::-moz-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1:-ms-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c1::placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

<div
  className="c0"
>
  <input
    className="c1"
    type="warning"
  />
</div>
`;

exports[`Input should return a with prefix and postfix Input 1`] = `
.c1 {
  border-left: 1px solid;
  border-radius: 8px 0 0 8px;
  padding: 0.5rem 1rem;
  margin: 0;
  font-size: 1.25rem;
  line-height: 1.75rem;
  background-color: white;
  border-top: 1px solid;
  border-bottom: 1px solid;
  border-color: rgb(209 213 219);
  color: black;
}

.c3 {
  border-right: 1px solid;
  border-radius: 0 8px 8px 0;
  padding: 0.5rem 1rem;
  margin: 0;
  font-size: 1.25rem;
  line-height: 1.75rem;
  background-color: white;
  border-top: 1px solid;
  border-bottom: 1px solid;
  border-color: rgb(209 213 219);
  color: black;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c2 {
  display: inline-block;
  min-width: 0;
  width: 100%;
  padding: 0.5rem 1rem;
  margin: 0;
  font-size: 1.25rem;
  line-height: 1.75rem;
  -webkit-transition: all .2s cubic-bezier(0.4,0,0.2,1);
  transition: all .2s cubic-bezier(0.4,0,0.2,1);
  background-color: white;
  border-width: 1px;
  border-style: solid;
  border-color: rgb(209 213 219);
  color: black;
  outline: none;
  border-radius: 0;
}

.c2:hover {
  border-color: #3B74F3;
}

.c2:focus {
  border-color: #3B74F3;
  -webkit-filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
  filter: drop-shadow(0 1px 2px rgb(0 0 0 / 0.1)) drop-shadow(0 1px 1px rgb(0 0 0 / 0.06));
}

.c2::-webkit-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c2::-moz-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c2:-ms-input-placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

.c2::placeholder {
  color: black;
  -webkit-filter: opacity(40%);
  filter: opacity(40%);
}

<div
  className="c0"
>
  <div
    className="c1"
    prefix="https://"
    type="base"
  >
    https://
  </div>
  <input
    className="c2"
    prefix="https://"
    type="base"
  />
  <div
    className="c3"
    prefix="https://"
    type="base"
  >
    .com
  </div>
</div>
`;
